### PR TITLE
Improve accuracy of vehicle fuel usage readouts

### PR DESCRIPTION
**Pull request by nornagon**
#### Summary

```SUMMARY: Interface "Vehicle fuel usage indicators now take into account engine load and time of day"```

#### Purpose of change
The fuel usage indicators in the vehicle overview screen currently show the _maximum_ usage (i.e. assuming the engine is at maximum power usage, and the solar panels are at maximum generation capacity). This PR changes those indicators to show the _current_ usage, taking into account weather & time of day in calculating solar panel output, and using the actual amount of fuel used in the last tick when factoring in fuel usage.

#### Describe the solution
I've refactored the various power calculation methods in vehicle to be more clear about what they compute. There's some duplication in the solar charging code still, because of the way the `vehicle::update_time` method integrates over the time since it was last updated.

I don't take into account reactors in this code path (but neither did the old code path). Since reactors currently produce instantaneous power rather than continuous power, it's challenging to include them.

I'm still not sure if these final calculations are 100% accurate, but they're a LOT more accurate than they were before (e.g. not claiming that solar panels are charging the batteries in the middle of the night).

#### Describe alternatives you've considered
Perhaps the indicators _should_ show nominal capacity rather than current? If so, then I think the UI should be a lot more clear about that—as is, the UI definitely makes it look like it's the _current_ usage, so I've tried to make the numbers match that.

#### Additional context
Solar car charging at 6am
Cataclysm__Dark_Days_Ahead_-_0_D-7186-g1c47e62a91-dirty

Solar car charging at midday
Cataclysm__Dark_Days_Ahead_-_0_D-7186-g1c47e62a91-dirty

Solar car driving at night
Cataclysm__Dark_Days_Ahead_-_0_D-7186-g1c47e62a91-dirty

Cruising in a van
Cataclysm__Dark_Days_Ahead_-_0_D-7186-g1c47e62a91-dirty

Idling in a van
Cataclysm__Dark_Days_Ahead_-_0_D-7186-g1c47e62a91-dirty

### CODE REVIEW

**Comment by nornagon** [comment]
Looks like I messed up a bunch of the vehicle efficiency tests 😅I'll take a look soon!

**Comment by nornagon** [comment]
hm... i'm not sure why the tests are failing on travis still. they pass locally and i don't see anything nondeterministic in them... any insights? could travis have inadvertently built the wrong code?

**Comment by nornagon** [comment]
I got [Chesterton's Fence](https://www.chesterton.org/taking-a-fence-down/)'d, didn't realise the alternator load computation was actually updating a class variable. Tests _actually_ pass locally now! Hoping they also pass on CI.

**Comment by nornagon** [comment]
NB. this change doesn't include electricity from wind turbines or water wheels. Planning to add those in a followup PR.

File: src/vehicle.cpp [hunk 1]
```diff
@@ -4259,19 +4263,49 @@ int vehicle::total_reactor_epower_w() const
...
+        int alternators_power = 0;
+        for( size_t p = 0; p < alternators.size(); ++p ) {
+            if( is_alternator_on( p ) ) {
+                alternators_power += part_vpower_w( alternators[p] );
+            }
+        }
+        alternator_load =
+            engine_vpower
+            ? 1000 * ( abs( alternators_power ) + abs( extra_drag ) ) / engine_vpower
+            : 0;
```
Comment (anothersimulacrum) [inline src/vehicle.cpp]: I'm not sure there's much benefit to having this as a ternary anymore, it would probably look better as an if/else statement.

**Pull request merged**